docs: note inotify limits for scale test (#360)
kubernetes-sigs/node-readiness-controller: docs/TEST_README.md

This section tests how the controller handles new nodes being added to the cluster, simulating an autoscaler.

> **Note: inotify limits on Linux hosts**
> Each `kind` worker node runs its own systemd instance inside a container,
> and each consumes several `inotify` instances (journald, udev, service
> watchers, etc.). The default Linux limit (`fs.inotify.max_user_instances`,
> often 128) is shared across your entire user session — desktop
> environment, editors, browsers, and other file-watching tools all draw

> > from the same pool. On a typical dev machine, this budget can already be

Reviewer: ```suggestion
> from the same pool. On some dev environments, this budget can already be
```

Is this specific to any Unix distro? I haven't experienced such issues for scaling more kind nodes in test environments. Also we use kwok for large scale testing (ref #284 )

Author: Its not really specific to any distro , it depends with inotify budget configured on your machine by default like for fedora max instances were set as 128 as default . but i tend to believe it depends with kernel version cause some other  distros ship with a higher configuration (even fedora the one im on) can ship with higher default instances set

> significantly consumed before any kind nodes start, making it easy to
> hit `Failed to allocate manager object: Too many open files` when

> > scaling to 3-4+ worker nodes.

Reviewer: ```suggestion
> scaling to more worker nodes.
```
